### PR TITLE
fix: Eager connect based on connector

### PR DESCRIPTION
**Pull request by RabbitDoge**
[ ] Before opening a pull request, please read the [contributing guidelines](https://github.com/pancakeswap/pancake-frontend/blob/master/CONTRIBUTING.md) first
[ ] If your PR is work in progress, open it as `draft`
[ ] Before requesting a review, all the checks need to pass
[ ] Explain what your PR does

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,7 +34,7 @@
     "@binance-chain/bsc-connector": "^1.0.0",
     "@crowdin/crowdin-api-client": "^1.10.4",
     "@ethersproject/abi": "^5.0.7",
-    "@pancakeswap-libs/uikit": "^0.12.0",
+    "@pancakeswap-libs/uikit": "../tmp/packages/pancakeswap-libs-uikit-v0.12.01111.tgz",
```
Comment (hachiojidev) [inline package.json:37]: Oops

File: src/hooks/useAuth.ts [hunk 2]
```diff
@@ -1,19 +1,22 @@
...
+  const login = useCallback((connectorID: ConnectorNames) => {
+    const connector = connectorsByName[connectorID]
     if (connector) {
       activate(connector, (error: Error) => toastError(error.name, error.message))
     } else {
       toastError("Can't find connector", 'The connector config is wriong')
     }
-  }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
```
Comment (hachiojidev) [inline src/hooks/useAuth.ts:19]: Is it activate or toastError causing the re-run?

**Review by hachiojidev — CHANGES_REQUESTED** [review]

File: src/hooks/useAuth.ts (hunk 2 above)
Comment (RabbitDoge) [reply src/hooks/useAuth.ts:19]: yes

File: package.json (hunk 1 above)
Comment (RabbitDoge) [reply package.json]: was waiting for the v13 release

**Review by hachiojidev — APPROVED** [review]

**Pull request merged**